src/main/java/org/traccar/reports/RouteReportProvider.java

```diff
@@ -66,7 +69,13 @@ public Collection<Position> getObjects(long userId, Collection<Long> deviceIds,
         }
         return result;
     }
+    private final Map<String, Integer> uniqueSheetsTitle = new HashMap<>();
 
+    private String getUniqueSheetName(String key) {
+        AtomicInteger atomic = new AtomicInteger(uniqueSheetsTitle.getOrDefault(key, -1));
+        uniqueSheetsTitle.put(key, atomic.incrementAndGet());
+        return (uniqueSheetsTitle.get(key) > 0 ? key + "-" + uniqueSheetsTitle.get(key) : key);
```

Comment:
Why do you need `AtomicInteger`? Also I think the numbers should start with 1 and not -1. So basically if there's 1 device, we don't show. If more than 1, we do "Name", "Name-2", "Name-3" etc.